src/lib/validation/validate.ts

```diff
@@ -74,6 +74,12 @@ export const getFormErrors = async (
   }
 };
 
+function isScalarValue(obj: IValues | Value | Values): obj is Value {
```

Comment:
Isn't this just a primitive (more common name) check? Can't we simplify this too: `const isPrimitive = (value) => value !== Object(value)`?

Reply:
the point of this function is that it provides type narrowing without casting it be a particular type, otherwise there are type errors due to the arbitrary JSON structure of the submission data.

Reply:
I see the purpose of the function here but I still have concerns about the confusing naming (primitive should be name here) and its implementation. The "temporary" nature doesn't not change that aspect. Also: being a todo I think it should be either commented or an issue should be created for it.

Reply:
There is actually a meaningful difference - scalar is the opposite from compounded (like arrays, objects...) and is intended to refer to actual, concrete values. From a mathematical point of view this is more correct, since primitives in JS also include symbols, which is most definitely not supported as a value.

I've created #27 to keep track of this.

Reply:
This pull requests shouldn't have been merged without approval. Independently of the semantic discussion whether we should adhere to math or js concepts. I have some concerns regarding the implementation which should have been resolved through the proper process.

Reply:
I merged it because it's fixing the  (reportedly) broken build chain, which was blocking any other feature and preventing us from using the formio utils which are known to have a correct implementation and our own implementation was undeniably insufficient

I shall adress the remaining comments tomorrow, either through justification as a comment or a follow up PR to correct them.

Reply:
Renamed the function to `isSingleValue` because that's essentially what the type predicate expresses too. As added in the comments, this is still not 100% correct but that requires tweaking the Value types/interfaces extensively. I'm really thinking we should be able to map the static formio component types to the matching value types and use some typescript generics magic get specific types instead of having to deal with mismatching types (e.g. boolean for textfield does not make any sense).